public/app/features/plugins/sandbox/distortion_map.ts

```diff
@@ -456,3 +460,48 @@ function distortDocument(distortions: DistortionMap) {
     }
   }
 }
+
+async function distortMonacoEditor(distortions: DistortionMap) {
+  // We rely on `monaco` being instanciated inside `window.monaco`.
+  // this is the same object passed down to plugins using monaco editor for their editors
+  // this `window.monaco` is an instance of monaco but not the same as if we
+  // import `monaco-editor` directly in this file.
+  // Short of abusing the `window.monaco` object we would have to modify grafana-ui to export
+  // the monaco instance directly in the ReactMonacoEditor component
+  const monacoEditor: Monaco = Reflect.get(window, 'monaco');
```

Comment:
Something for the future... I believe that if we move monaco editor to load via ESM no globals are created which would make this undefined. I don't think it should block us from merging this but certainly something we need to consider. @aangelisc looping you in so you're aware.

Reply:
I think once we're ready to move Monaco to ESM we can move the `monaco-kusto` package to also load via ESM which should *hopefully* make our problems go away.